src/components/Orderbook.tsx

```diff
@@ -1,196 +1,414 @@
+          order.minerFeeContribution.replace('.', '').toLowerCase().includes(searchVal) ||
+          order.bondValue.replace('.', '').toLowerCase().includes(searchVal) ||
+          order.orderId.toLowerCase().includes(searchVal)
+        )
+      })
+      .map((order) => ({
+        ...order,
+        id: `${order.counterparty}_${order.orderId}`,
+      }))
+
+    return { nodes }
+  }, [orders, search])
+
+  return (
+    <div className={styles.orderbookContainer}>
+      <div className={styles.titleBar}>
+        <div className="d-flex justify-content-center align-items-center gap-2">
+          <rb.Button
+            className={styles.refreshButton}
+            variant={settings.theme}
+            onClick={() => {
+              if (isLoadingRefresh) return
+
+              setIsLoadingRefresh(true)
+
+              const abortCtrl = new AbortController()
+              refresh(abortCtrl).finally(() => {
+                // as refreshing is fast most of the time, add a short delay to avoid flickering
+                setTimeout(() => setIsLoadingRefresh(false), 250)
+              })
+            }}
+          >
+            {isLoadingRefresh ? (
+              <rb.Spinner as="span" animation="border" size="sm" role="status" aria-hidden="true" />
+            ) : (
+              <Sprite symbol="refresh" width="24" height="24" />
+            )}
+          </rb.Button>
+          <div className="small">
+            {t('orderbook.text_orderbook_summary', {
+              counterpartyCount,
+              orderCount: orders.length,
+            })}
+            {search && (
+              <>
+                <br />(
+                {t('orderbook.text_orderbook_filter_count', {
+                  filterCount: tableData.nodes.length,
+                })}
+                )
+              </>
+            )}
           </div>
-          <rb.Table striped bordered hover variant={settings.theme} responsive>
-            <thead>
-              <tr>
-                {Object.values(headingMap).map((header, index) => (
-                  <th key={`header_${index}`}>{header.heading}</th>
-                ))}
-              </tr>
-            </thead>
-            <tbody>
-              {orders.map((order, index) => (
-                <tr key={`order_${index}_${order.orderId}`}>
-                  {columns.map((propName) => (
-                    <td key={propName}>
-                      {headingMap[propName] && headingMap[propName].render !== undefined
-                        ? headingMap[propName].render!(order[propName])
-                        : order[propName]}
-                    </td>
-                  ))}
-                </tr>
-              ))}
-            </tbody>
-          </rb.Table>
         </div>
-      )}
-    </>
+        <div>
+          <rb.Form.Group controlId="search">
+            <rb.Form.Label className="m-0 pe-2 d-none">{t('orderbook.label_search')}</rb.Form.Label>
+            <rb.Form.Control
```

Comment:
Any value in disabling search while the orders are refreshed?

Reply:
Done.